### DISCUSSION TITLE
Shared nodes

### DISCUSSION THREAD

**Discussion by hlolve** [idena-network/idena-desktop#821] [General]
Make idena-desktop possible to use shared nodes.
Similar to the use in browsers, but it is better to use idena-desktop for PC
So enable access for any account using any shared node.

**Comment by rioda-org** [comment]
It would be great if the App could detect that there are sync issues near validation time, and it could auto rent/connect to the closest node.

**Comment by ltraveler** [comment]
The better idea would be to save all API keys per epoch in the blockchain and in case if the client have node conection problem some minutes before the validation ceremony the blockchain will provide an additional API key from the closest node.

**Reply by Zen-44** [reply]
This would make them publicly available.

**Reply by ltraveler** [reply]
That make sense. Each of the key must be encrypted and the specific key could decrypted only buy user who bought it and in case of providing additional key some minutes bofore the validation ceremony the blockchain could randomly provide some extra backup key/keys from different node/nodes as a backup option of course that they should be applied automatically.

**Reply by Zen-44** [reply]
You can't let the blockchain decrypt those keys, because that would also make them publicly available. The decryption would have to come from some central entity.

**Reply by ltraveler** [reply]
One of the option to change authentification method. Instead of using keys, it could be a new transaction in the blockchain to sell API key. So, if there is a transaction like that during the epoch the identity automatically get an access to any shared node from the market. In that case the price for renting should be constant and after validation all accumulated coins would be spread between shared node operators based on the opinion of users after validation and it must be obligatory for user to make a feed back before to send answers in the long session. Actually with that method we can make a balance loading between the whole network of shared nodes. Of course to be in the market the candidate must to prove the minimum required power of the server it could be a unixbenchmark test results and the candidate must to pass some epochs as trial period to prove the quality of the server, during that period candidate is not getting any reward. It's like a period to get a minimum required reputation.

**Comment by kapo89star** [comment]
This is great idea, I saw so many people failed because before validation node is not sync.

**Comment by hlolve** [comment]
Thanks shared node operators for participate

I see shared nodes perfect as backup, is an option we are not using

If sync is not working in local or remote can use shared as last resort

Users think Idena network is failing but is their connection that can't handle the node, but still possible using shared 

If a user do not want to use browser for passwords security reason we have the app

Also will help to troubleshoot connections for browser configs

And also we are not depending of browsers, only use for smartphones or tablets
